File: budget_buddy/src/main/java/com/example/budget_buddy/dto/ExpenseDto.java
package com.example.budget_buddy.dto;

import java.math.BigDecimal;
import java.util.Date;

import com.example.budget_buddy.model.Expense;

public class ExpenseDto {
    private Integer id;
    private String expenseName;
    private BigDecimal cost;
    private String category;
    private String description;
    private Date expenseDateTime;

    public ExpenseDto() {
    }

    public ExpenseDto(Expense expense) {
        this.id = expense.getId();
        this.expenseName = expense.getExpenseName();
        this.cost = expense.getCost();
        this.category = expense.getCategory();
        this.description = expense.getDescription();
        this.expenseDateTime = expense.getExpenseDateTime();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getExpenseName() {
        return expenseName;
    }

    public void setExpenseName(String expenseName) {
        this.expenseName = expenseName;
    }

    public Date getExpenseDateTime() {
        return expenseDateTime;
    }

    public void setExpenseDateTime(Date expenseDateTime) {
        this.expenseDateTime = expenseDateTime;
    }

    public BigDecimal getcost() {
        return cost;
    }

    public void setcost(BigDecimal cost) {
        this.cost = cost;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: budget_buddy_android/app/src/main/java/com/example/budget_buddy_android/dto/ExpenseDto.kt
package com.example.budget_buddy_android.dto

import com.example.budget_buddy_android.models.Expense
import java.math.BigDecimal
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class ExpenseDto(
    val expenseName: String,
    val cost: BigDecimal,
    val category: String,
    val description: String,
    val expenseDateTime: String
) {
    constructor(expense: Expense) : this(
        expenseName = expense.expenseName,
        cost = expense.cost,
        category = expense.category,
        description = expense.description,
        expenseDateTime = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.getDefault()).format(expense.expenseDateTime)
    )
    constructor():this(
        expenseName = "",
        cost = BigDecimal(0),
        category = "",
        description = "",
       expenseDateTime = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.getDefault()).format(
           Date()
       )
    )
}